opensearch/src/main/java/org/opensearch/sql/opensearch/data/value/OpenSearchExprValueFactory.java

```diff
@@ -338,39 +353,97 @@ private ExprValue createOpenSearchDateType(Content value, ExprType type) {
     return new ExprTimestampValue((Instant) value.objectValue());
   }
 
-  private ExprValue parseStruct(Content content, String prefix) {
+  /**
+   * Parse struct content.
+   * @param content Content to parse.
+   * @param prefix Prefix for Level of object depth to parse.
+   * @param supportArrays Are non-object arrays supported.
+   * @return Value parsed from content.
+   */
+  private ExprValue parseStruct(Content content, String prefix, boolean supportArrays) {
     LinkedHashMap<String, ExprValue> result = new LinkedHashMap<>();
     content.map().forEachRemaining(entry -> result.put(entry.getKey(),
         parse(entry.getValue(),
             makeField(prefix, entry.getKey()),
-            type(makeField(prefix, entry.getKey())))));
+            type(makeField(prefix, entry.getKey())), supportArrays)));
     return new ExprTupleValue(result);
   }
 
   /**
-   * Todo. ARRAY is not completely supported now. In OpenSearch, there is no dedicated array type.
-   * <a href="https://opensearch.org/docs/latest/opensearch/supported-field-types/nested/">docs</a>
-   * The similar data type is nested, but it can only allow a list of objects.
+   * Parse array content. Can also parse nested which isn't necessarily an array.
+   * @param content Content to parse.
+   * @param prefix Prefix for Level of object depth to parse.
+   * @param supportArrays Are non-object arrays supported.
+   * @param type Type of content parsing.
+   * @return Value parsed from content.
    */
-  private ExprValue parseArray(Content content, String prefix) {
+  private ExprValue parseArray(
+      Content content,
+      String prefix,
+      ExprType type,
+      boolean supportArrays
```

Comment:
`supportArrays` seems like a weird parameter in a `parseArrays` function.  You could call this `isNonObjectArraySupported` or something like that...